src/validation/__tests__/UniqueVariableNames.js

```diff
@@ -0,0 +1,45 @@
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import { describe, it } from 'mocha';
+import { expectPassesRule, expectFailsRule } from './harness';
+import {
+  UniqueVariableNames,
+  duplicateVariableMessage,
+} from '../rules/UniqueVariableNames';
+
+
+function duplicateVariable(name, l1, c1, l2, c2) {
+  return {
+    message: duplicateVariableMessage(name),
+    locations: [ { line: l1, column: c1 }, { line: l2, column: c2 } ],
+  };
+}
+
+describe('Validate: Unique variable names', () => {
+
+  it('unique variable names', () => {
+    expectPassesRule(UniqueVariableNames, `
+      query A($x: Int, $y: String) { __typename }
+      query B($x: String, $y: Int) { __typename }
+    `);
+  });
+
+  it('duplicate variable names', () => {
+    expectFailsRule(UniqueVariableNames, `
+      query A($x: Int, $x: Int, $x: String) { __typename }
+      query B($x: String, $x: Int) { __typename }
```

Comment:
perhaps also include a test like `query C($x: Int, $x: Int) { __typename }` just to illustrate that *only* duplicating a variable name is still not ok